app/models/user.rb

```diff
@@ -549,14 +549,36 @@ def subscribed_commentable_media_with_comments
       .select { |m| m.commontator_thread.comments.any? }
   end
 
-  def media_latest_comments
-    media = subscribed_commentable_media_with_comments
-            .map do |m|
+  # Returns the media that the user has subscribed to and that have been
+  # commented on by somebody else (not by the current user). The order is
+  # given by the time of the latest comment by somebody else.
+  #
+  # Media that have not been commented on by somebody else than the current user,
+  # are not returned (!).
+  #
+  # For each medium, the following information is stored:
+  # - the medium itself
+  # - the thread of the medium
+  # - the latest comment by somebody else than the current user
+  # - the latest comment by any user (which might include the current user)
+  def subscribed_media_with_latest_comments_not_by_creator
+    media = subscribed_commentable_media_with_comments.map do |m|
+      thread = m.commontator_thread
+      comments = thread.comments
+      next unless comments
```

Comment:
`comments` is an array, and empty arrays are truthy in Ruby. You might want to replace this by 

```ruby
next if comments.blank?
```